common/defs.go

```diff
@@ -82,6 +89,76 @@ type PortUpdateMessage struct {
 	Port uint64 `json:"port"`
 }
 
+type Endpoint struct {
+	CidrStr string `json:"cidr,omitempty"`
+	// TODO this can be collapsed into Cidr but needs
+	// work on JSON marshaller/unmarshaller to do that.
+	Cidr      net.IPNet `json:"-"`
+	TenantID  uint64    `json:"tenant_id,omitempty"`
+	TenantExternalID string  `json:"tenant_external_id,omitempty"`
+	SegmentID uint64    `json:"segment_id,omitempty"`
+	SegmentExternalID string `json:"segment_external_id,omitempty"`
+}
+
+const (
+	PolicyDirectionIngress = "ingress"
+	PolicyDirectionEgress  = "egress"
+)
+
+type PortRange [2]uint
+
+type Rule struct {
+	Protocol   string
+	Ports      []uint
+	PortRanges []PortRange
+	IcmpType   uint
+	IcmpCode   uint
+	Stateful   bool
+}
+
+type Rules []Rule
+
+// Metadata attached to entities for various external environments like openstack/kybernetes
+type Tag struct {
+	Key   string
+	Value string
+}
+
+type Policy struct {
+	Direction   string     `json:"direction,omitempty"`
+	Description string     `json:"description,omitempty"`
+	Name        string     `json:"name,omitempty"`
```

Comment:
Please note that there are certain hard limits on the length of the policy name that we can accept. We are currently encoding the policy name in the iptables chain names. The current limit is only 6 (six!) characters!

We will change this over time to be able to accommodate a little more, but iptables always enforces a hard limit somewhere.

However, instead of limiting the name on input here, the better solution is probably in the policy manager, or even the policy agent, to shorten the policy name to some unique, short value. We might even have to switch away from using the policy name in the chain name and instead use some sort of short unique ID, or a number.

Anyway, I just want to put this out here, so that we are aware of this.